### PR TITLE
Fix initializing auth in auth-table-based-service (fixes #288)

### LINKED ISSUE #288: Unable to initialize auth when using DSE backend
When starting up in developer mode with DSE auth-table-based-service is unable to create the keyspace and table since there isn't a user in the QueryState. This works in both C* 3.11 and C* 4.0 because there's no user check when executing the query.

### PR DESCRIPTION
**Pull request by dougwettlaufer**

### CODE REVIEW

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- testing/src/test/java/io/stargate/it/http/HealthCheckerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6354980&bid=20633065)

File: persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java [hunk 1]
```diff
@@ -348,7 +348,12 @@ protected void loginInternally(io.stargate.db.AuthenticatedUser user) {
 
     private Single<QueryState> newQueryState() {
       if (clientState.getUser() == null) {
-        return Single.just(new QueryState(clientState, UserRolesAndPermissions.ANONYMOUS));
+        // This is here to maintain consistent behavior with C* 3.11 and C* 4.0 since DSE requires
```
Comment (tomekl007) [inline persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java:351]: What if someone from the outside will know this `anonymous` user name?

Comment (dougwettlaufer) [reply persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java:351]: The only time there's no user on the clientState is during internal calls, elsewhere all requests go through an authentication (e.g. cql, rest, and graphql).

Comment (EricBorczuk) [reply persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java:351]: So there is no way for a requester to log in as `ANONYMOUS_USER`? (does the role have `with login = false` or smth)

**Review by EricBorczuk — APPROVED** [review]

File: persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java (hunk 1 above)
Comment (dougwettlaufer) [reply persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java:351]: Well if they somehow logged in with the anonymous user then the clientState would have a user and you wouldn't drop into this block.

**Pull request merged**